### DISCUSSION TITLE
CNAME support for provider 'ionos'

### DISCUSSION THREAD

**Discussion by markusm7** [go-acme/lego#2280] [Q&A] [answered]
Im currently using lego as acme client in traefik and works nicely for all my usecases.
But now im trying to create certificates via CNAME pointing to a domain I own and is part of the ionos DNS zone.
But im getting the error `ionos: no matching zone found for domain` - at first I was thinking there must be something wrong with my setup, tried different `CNAME` records, different DNS resolvers and so on. Every issue I found online was always related to some misconfiguration either in traefik/lego or DNS problems.
As the api was working and returning the correct zones (also including the correct one for my domain) I really had no idea what im doing wrong, as lego should follow `CNAME` records correctly (according to documentation)

But going through the code of the ionos I found the problem:
https://github.com/go-acme/lego/blob/0da0942081a1666a73b4f2f3f35328e9ae28af0f/providers/dns/ionos/ionos.go#L116-L120

So my question: Is this going to be implemented as the "TODO(@ldez)" comment is suggesting? And if yes, how long would it take to have a new lego/traefik release?

If that discussion should be an issue let me know! I wasn't sure where to put it as this isn't really a bug report nor a real feature request as its something the tool should can as promoted and already does for other providers than `ionos`

**Comment by ldez** [comment, marked as answer]
Hello,

I can implement this but I need someone with a IONOS account to test the PR.

If I create a PR, can you test it?

EDIT: https://github.com/go-acme/lego/pull/2281

**Reply by markusm7** [reply]
I guess I would have to install lego directly on a host (or maybe locally would suffice I reckon) correct?
If you can tell me me how I can do that and what calls/commands I need to test for you I'm willing to do that

EDIT: nvm, saw the commands in the PR - I'll let you know

**Reply by ldez** [reply]
you can test it locally as the DNS challenge is not dependent on the env (it only relies on DNS).
